### ISSUE TITLE
Incomplete code sample causes CCE in `VariableDeclarationNode.java`

### ISSUE THREAD

**Issue by rdulmina** [ballerina-platform/ballerina-lang#34486]
**Description:**
Below sample causes CCE in `VariableDeclarationNode.java` (Note the extra comma after `g` int the error binding pattern)
```bal
public function main() {
    [error, boolean...] [error(g,), ...j] = [error("error!"), true];
}
```
<details>
  <summary>Stack trace.(Click to expand)</summary>

```
[Error - 10:52:46 AM] Operation 'ballerinaDocument/executorPositions' failed! {uri: '/home/dulmina/Documents/test.bal', error: 'class io.ballerina.compiler.syntax.tree.IndexedExpressionNode cannot be cast to class io.ballerina.compiler.syntax.tree.TypedBindingPatternNode (io.ballerina.compiler.syntax.tree.IndexedExpressionNode and io.ballerina.compiler.syntax.tree.TypedBindingPatternNode are in unnamed module of loader 'app')'}
java.lang.ClassCastException: class io.ballerina.compiler.syntax.tree.IndexedExpressionNode cannot be cast to class io.ballerina.compiler.syntax.tree.TypedBindingPatternNode (io.ballerina.compiler.syntax.tree.IndexedExpressionNode and io.ballerina.compiler.syntax.tree.TypedBindingPatternNode are in unnamed module of loader 'app')
	at io.ballerina.compiler.syntax.tree.VariableDeclarationNode.typedBindingPattern(VariableDeclarationNode.java:45)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeBuilder.transform(BLangNodeBuilder.java:2770)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeBuilder.transform(BLangNodeBuilder.java:472)
	at io.ballerina.compiler.syntax.tree.VariableDeclarationNode.apply(VariableDeclarationNode.java:67)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeBuilder.generateAndAddBLangStatements(BLangNodeBuilder.java:4960)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeBuilder.transform(BLangNodeBuilder.java:1541)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeBuilder.transform(BLangNodeBuilder.java:472)
	at io.ballerina.compiler.syntax.tree.FunctionBodyBlockNode.apply(FunctionBodyBlockNode.java:59)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeBuilder.populateFunctionNode(BLangNodeBuilder.java:1448)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeBuilder.createFunctionNode(BLangNodeBuilder.java:1428)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeBuilder.transform(BLangNodeBuilder.java:1406)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeBuilder.transform(BLangNodeBuilder.java:472)
	at io.ballerina.compiler.syntax.tree.FunctionDefinitionNode.apply(FunctionDefinitionNode.java:71)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeBuilder.transform(BLangNodeBuilder.java:606)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeBuilder.transform(BLangNodeBuilder.java:472)
	at io.ballerina.compiler.syntax.tree.ModulePartNode.apply(ModulePartNode.java:54)
	at org.wso2.ballerinalang.compiler.parser.BLangNodeBuilder.accept(BLangNodeBuilder.java:505)
	at io.ballerina.projects.DocumentContext.compilationUnit(DocumentContext.java:111)
	at io.ballerina.projects.ModuleContext.compileInternal(ModuleContext.java:424)
	at io.ballerina.projects.ModuleCompilationState$3.compile(ModuleCompilationState.java:100)
	at io.ballerina.projects.ModuleContext.compile(ModuleContext.java:386)
	at io.ballerina.projects.PackageCompilation.compileModulesInternal(PackageCompilation.java:213)
	at io.ballerina.projects.PackageCompilation.compileModules(PackageCompilation.java:198)
	at io.ballerina.projects.PackageCompilation.compile(PackageCompilation.java:114)
	at io.ballerina.projects.PackageCompilation.from(PackageCompilation.java:103)
	at io.ballerina.projects.PackageContext.getPackageCompilation(PackageContext.java:214)
	at io.ballerina.projects.Package.getCompilation(Package.java:141)
	at org.ballerinalang.langserver.extensions.ballerina.document.ExecutorPositionsUtil.getExecutorPositions(ExecutorPositionsUtil.java:75)
	at org.ballerinalang.langserver.extensions.ballerina.document.BallerinaDocumentService.lambda$executorPositions$15(BallerinaDocumentService.java:392)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1700)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1692)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:177)
```
</details>

**Comment by SasinduDilshara** [comment]
Seems like same thing happen in 
```
public function main() {
    [error, boolean...] [error(g,(), ...j] = [error("error!"), true];
}
```

**Comment by lochana-chathura** [comment]
Will be fixed with #38064
